java/employee-scheduling/src/main/java/org/acme/employeescheduling/domain/Shift.java

```diff
@@ -89,6 +96,41 @@ public void setEmployee(Employee employee) {
         this.employee = employee;
     }
 
+    @JsonIgnore
+    public boolean isOverlappingWith(Collection<LocalDate> dates) {
+        return !getOverlappingDatesForShift(dates).isEmpty();
+    }
```

Comment:
Although this is a simple way of implementing the requirement, it is also an anti-pattern. 

Let us not teach our users that abandoning incrementality and turning everything into collection processing is the right way of doing things; they already think that anyway. There is a simple rule of thumb when working with constraint streams: if you see a collection or a for-loop anywhere, you're probably doing it wrong.

You have to process each of `LocalDate` individually. You can use `flatten(...)` to get it out of the `Shift`.